### PR TITLE
python3Packages.wxPython_4_0: fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Also refactored wxGTK30 and added myself as a maintainer.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures on macos 10.15 (but that's a fraction of what's rebuilt for linux)

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures on nixos x86_64

File: pkgs/development/libraries/wxwidgets/wxGTK30.nix [hunk 1]
```diff
@@ -1,29 +1,37 @@
 { lib
 , stdenv
+, expat
 , fetchFromGitHub
 , gst_all_1
+, withGtk2 ? true
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/wxwidgets/wxGTK30.nix:6]: Can we drop this?

Comment (wegank) [reply pkgs/development/libraries/wxwidgets/wxGTK30.nix:6]: `wxGTK30` defaults to `withGtk2 = true` in `pkgs/top-level/all-packages.nix`, so I guess it's not immediate.

Comment (SuperSandro2000) [reply pkgs/development/libraries/wxwidgets/wxGTK30.nix:6]: GTK2 is pretty EOL and already removed from many places. I would guess that wxGTK30 is used by accident and really we want to use the gtk3 variant in almost all places.

Comment (wegank) [reply pkgs/development/libraries/wxwidgets/wxGTK30.nix:6]: In the meantime, I'm pretty sure that we can drop GTK2 from wxGTK31 after #194682.

**Pull request merged**